### PR TITLE
Upgrade to torchmetrics 0.11

### PR DESCRIPTION
**Pull request by wistuba**
Account for interface changes in torchmetrics 0.11.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/renate/benchmark/experiment_config.py [hunk 1]
```diff
@@ -361,5 +359,6 @@ def lr_scheduler_fn(
     raise ValueError(f"Unknown scheduler `{learning_rate_scheduler}`.")
 
 
-def metrics_fn() -> Dict:
-    return {"accuracy": Accuracy()}
+def metrics_fn(num_outputs: int) -> Dict:
+    task = "binary" if num_outputs == 2 else "multiclass"
```
Comment (lballes) [inline src/renate/benchmark/experiment_config.py:363]: I don't think this will work. IIUC, the binary accuracy in `torchmetrics` expects scalar predictions.

File: examples/getting_started/renate_config.py [hunk 2]
```diff
@@ -93,7 +93,7 @@ def buffer_transform() -> Callable:
 
 
 def metrics_fn() -> Dict:
-    return {"accuracy": Accuracy()}
+    return {"accuracy": Accuracy(task="multiclass", num_classes=10)}
```
Comment (lballes) [inline examples/getting_started/renate_config.py:96]: We need to be very careful with this new `torchmetrics` interface. It caused me some troubles in another project. By default it uses a `"macro"` average for multiclass accuracy, i.e., it computes average accuracy for each label, then averages those. We probably want the `"micro"` average, right?

**Review by lballes — CHANGES_REQUESTED** [review]

File: examples/getting_started/renate_config.py (hunk 2 above)
Comment (wistuba) [reply examples/getting_started/renate_config.py:96]: That is correct but we are using Accuracy which uses "micro": https://torchmetrics.readthedocs.io/en/stable/classification/accuracy.html#accuracy
I've initially used the MulticlassAccuracy and then the tests failed because of the macro averaging. Now they pass, so things should work as intended.

File: src/renate/benchmark/experiment_config.py (hunk 1 above)
Comment (wistuba) [reply src/renate/benchmark/experiment_config.py:363]: good point. do we want to use `num_outputs==1` and account for this change in `loss_fn` and `model_fn`?

File: examples/getting_started/renate_config.py (hunk 2 above)
Comment (wistuba) [reply examples/getting_started/renate_config.py:96]: ok, I've changed everything to multiclass and added the `average` argument

**Review by lballes — APPROVED** [review]
